src/identity-utils.ts

```diff
@@ -17,74 +30,101 @@ export type IParseCachedIdentityResponse = (
     identityMethod: string,
     knownIdentities: IKnownIdentities,
     fromCachedIdentity: boolean
-) => void
+) => void;
 
 export interface IKnownIdentities extends UserIdentities {
     device_application_stamp?: string;
 }
 
 export interface ICachedIdentityCall {
+    // https://go.mparticle.com/work/SQDSDKS-6672
     responseText: string;
     status: number;
     expireTimestamp: number;
 }
 
+// https://go.mparticle.com/work/SQDSDKS-6568
+// Temporary adapter to convert the XMLHttpRequest response to the IIdentityResponse interface
+export const xhrIdentityResponseAdapter = (
+    possiblyXhr: XMLHttpRequest | IIdentityResponse
+): IIdentityResponse => {
+    if (possiblyXhr.hasOwnProperty('expireTimestamp')) {
+        // This is likely an IIdentityResponse object, just return it
+        return possiblyXhr as IIdentityResponse;
+    } else {
+        // This might be an actual xhr object, so we need to parse it
```

Comment:
instead of saying it *might* be an actual xhr object in this path, I think the comment should be more clear that it IS an xhr object going down this code path.  So maybe we should just say that.
```suggestion
        // If there is no `expireTimestamp`, then it is an XHR object and needs to be parsed.
```